database.go

```diff
@@ -71,6 +73,41 @@ type ConnectionConfig struct {
 	Params   string
 }
 
+/*
+ * The user specified url for connecting to a database. If nil, no url was provided by the user.
+ */
+type ConnectionURL struct {
+	URL *url.URL
+}
+
+func (v ConnectionURL) OverrideConnectionParams(connectionConfig *ConnectionConfig, driverName *string) {
+	if v != (ConnectionURL{}) {
+		u := v.URL
+		if u.Scheme != "" {
+			*driverName = u.Scheme
+		}
+		if u.Host != "" {
```

Comment:
Here and below, it seems like it would be a bit nicer for this to be a method on ConnectionConfig (the object being modified), e.g.

```
func (cc *ConnectionConfig) OverrideFromURL(u *url.URL) { }
```

Reply:
Thanks @awreece , this is a very good idea! I've done my best to implement it, please let me know if it fits your requirements.